documentation/DEVELOPER-GUIDELINES.md

```diff
@@ -61,6 +61,7 @@ In general, the [Rust API Guidelines](https://rust-lang.github.io/api-guidelines
     - see [this example](https://github.com/esp-rs/esp-hal/blob/df2b7bd8472cc1d18db0d9441156575570f59bb3/esp-hal/src/spi/mod.rs#L15)
     - e.g. `#[cfg_attr(feature = "defmt", derive(defmt::Format))]`
   - Implementations of common, but unstable traits (e.g. `embassy_embedded_hal::SetConfig`) need to be gated with the `unstable` feature.
+- Libraries depending on esp-hal, should disable the `rt` feature to avoid future compatibility concerns.
```

Comment:
IMHO this will need to be documented elsewhere, as this is esp-hal documentation. Since we'll need to teach third parties about esp-metadata-generated, we can probably start a new chapter here.

Reply:
As in the book? Because I documented in lib.rs too.

Reply:
I was thinking more about a new chapter in this file, but maybe it's enough to just document the feature in Cargo.toml instead.

Reply:
Done